drivers/serial/Kconfig.stm32

```diff
@@ -10,6 +10,7 @@ menuconfig UART_STM32
 	default n
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
+	select USE_STM32F0XX_HAL_USART if SOC_SERIES_STM32F0X
```

Comment:
Does the STM32F0 family use the HAL serial driver? I thought that all families use the LL serial api.

Reply:
F0 should use the same as all STM32 families

Reply:
Could we remove this line or should we do it in a separate PR?

Reply:
Should be a conversion of an oversight from F0 PR merged following merge of the PR that converted serial to full LL (Not quite clear, but you got my point).
To do it in a clean way, I propose an initial commit that remove culprit line in ext/hal/st/stm32cube/Kbuild.

Reply:
Acked

Reply:
I modeled the driver-kconfig files after the hal-Kconfig file that had;
`obj-$(CONFIG_SERIAL_HAS_DRIVER) += stm32f0xx/drivers/src/stm32f0xx_hal_uart.o`

Reply:
That was my point this line was an oversight following of F0 series introduction.
Migration from HAL to LL in serial driver has been merged during process of F0 review. It should have been detected at that point (my bad).
It would be nice if you could do a preliminary commit to your PR where you remove this line.

Reply:
@lowlander : Btw, just observed that initial Kbuild has been fixed with #4493.
So you could remove it safely after rebase